### DISCUSSION TITLE
Adding a "Timer" variable to control.

### DISCUSSION THREAD

**Discussion by Krutors** [Rem0o/FanControl.Releases#2659] [Ideas]
Hi, I'm opening this to know if it would be possible to incorporate a "Timer" variable into a control.

For example : I control the fans hooked up to my custom loop through FC, and have done a custom fan curve. The fans ramp up when a load comes in and get back down when the temps go below the threshold, but the liquid still need to be cooled.

Said condition would go as follows : 
After going above a certain temp, let the attached control run at xxx RPM for yyy seconds, regardless of the temp reported by the associated sensor.

Or maybe it's already possible to do something like this in FC as it is, in that case, you can disregard that post.

Have a nice day :)

**Comment by Rem0o** [comment]
Add something like this into your Mix fan curve:
![image](https://github.com/user-attachments/assets/156cfbb4-4472-4c67-b174-cc121f1a4755)

The trigger will hold the % up until 30 second has passed where the temp dropped below the Idle temp.

**Reply by Rem0o** [reply]
You can also use 1-way hysteresis with a high response time to get a similar effect, but the "trigger in the mix" method gives you the more precise control.

**Reply by Krutors** [reply]
> Add something like this into your Mix fan curve: ![image](https://private-user-images.githubusercontent.com/14118956/357115886-156cfbb4-4472-4c67-b174-cc121f1a4755.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0ODg5NTMsIm5iZiI6MTcyMzQ4ODY1MywicGF0aCI6Ii8xNDExODk1Ni8zNTcxMTU4ODYtMTU2Y2ZiYjQtNDQ3Mi00YzY3LWIxNzQtY2MxMjFmMWE0NzU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEyVDE4NTA1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzYzJhMDIyMTAyYTc5ZTMwZDQ5ZTc4Y2Q2ZDRlN2U5ZDYyMjgwOWY1ODU3N2VjYTk0MDFlNmExYmU3YzkxYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CaJTxpS99mkTs_OXp-7UBCHbC1ZedYjtIsay_2RqlwI)
> 
> The trigger will hold the % up until 30 second has passed where the temp dropped below the Idle temp.

First of all, thanks for the quick reply.

I thought about using the "response time" variable at first, but had disregarded it thinking it would also apply on the way up, wich I don't want. But if you sau it should work, I'll give it a try and report back :)
